example/src/main/java/com/stripe/example/view/CreditCardView.java

```diff
@@ -0,0 +1,560 @@
+        mSpaceInContainer = (Space) findViewById(R.id.space_in_container);
+        mIvCreditCardIcon = (ImageView) findViewById(R.id.iv_credit_card_icon);
+        mEtNumber = (EditText) findViewById(R.id.et_credit_card_number);
+        mEtExpiryDate = (EditText) findViewById(R.id.et_expiry_date);
+        mEtCvcNum = (EditText) findViewById(R.id.et_cvc_num);
+    }
+
+    private void initView() {
+        inflate(getContext(), R.layout.view_credit_card, this);
+
+        // TODO : could also do this with ButterKnife
+        bindViews();
+
+        mError = ERROR_NONE;
+
+        mColor = ContextCompat.getColor(getContext(), android.R.color.white);
+        mScrollView.setScrollingEnabled(false);
+
+        mScrollViewWidth = (int) (getScreenWidth(getContext()) * 0.65f);
+
+        mScrollView.getLayoutParams().width = mScrollViewWidth;
+        mEtNumber.getLayoutParams().width = mScrollViewWidth;
+        mSpaceInContainer.getLayoutParams().width = mScrollViewWidth;
+
+        mScrollToPosition = mScrollViewWidth;
+
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(new Date());
+
+        mEtNumber.setOnFocusChangeListener(new OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View view, boolean focused) {
+                if (focused) {
+                    scrollLeft();
+                }
+            }
+        });
+
+        mEtNumber.addTextChangedListener(new TextWatcher() {
+            boolean freeze = false;
+            int beforeStringLen = 0;
+
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int start, int count, int after) {
+                if (!freeze) {
+                    beforeStringLen = mEtNumber.getText().toString().length();
+                }
+            }
+
+            @Override
+            public void onTextChanged(CharSequence charSequence, int start, int before, int count) {
+                if (!freeze) {
+                    String curString = mEtNumber.getText().toString();
+                    int curStringLen = curString.length();
+
+                    mEtNumber.setTextColor(mColor);
+
+                    // reformat with spaces
+                    // TODO : add formatting for non-16 digit cards
+                    String cleanedString = curString.replace(" ", "");
+                    int cleanedStringLen = cleanedString.length();
+                    String formattedString = "";
+                    for (int i = 0 ; i < 4 ; i++) {
+                        if (cleanedStringLen > (i * 4) + 4) {
+                            formattedString += cleanedString.substring((i * 4), ((i * 4) + 4)) + " ";
+                        } else {
+                            formattedString += cleanedString.substring((i * 4), cleanedStringLen);
+                            break;
+                        }
+                    }
+                    freeze = true;
```

Comment:
Is this having the same affect as just having an atomic section? Or do you want the rest of the method to still happen if we switch threads?